main/src/ca/uwaterloo/flix/language/phase/Redundancy.scala

```diff
@@ -598,8 +598,14 @@ object Redundancy {
     case Expression.PutStaticField(_, exp, _, _, _, _) =>
       visitExp(exp, env0, rc)
 
-    case Expression.NewObject(_, _, _, _, _) =>
-      Used.empty
+    case Expression.NewObject(_, _, _, _, methods, _) =>
```

Comment:
I think it may be worthwhile to add a test for this. There are two things to check for: shadowed variables and unused variables.

Are we at the point where we could add a couple tests to `TestRedundancy.scala`?

Reply:
So I've added a couple of tests, but the shadowed variable test is failing with:

ShadowedVar.NewObject.01 - Expected an error of type ShadowedVar, but found class ca.uwaterloo.flix.language.errors.RedundancyError$UnusedVarSym.

I'm unsure whether that's because:

a) My test is wrong in some way
b) The test is exposing a bug in the code I've added to `Redundancy.scala`
c) We need to finish working through the remaining phases before this test can work.

Advice gratefully received!

Reply:
Now that my other test is actually a test, it is also failing, but with an error that makes more sense to me:

UnusedFormalParam.NewObject.01 - Expected an error of type UnusedFormalParam, but found class ca.uwaterloo.flix.language.errors.RedundancyError$UnusedVarSym.

What I'm not sure about is whether the right thing to do is change the test (because that's the right error) or change the code to generate a different error.

Reply:
`UnusedFormalParam` is only created via a call to `findUnusedFormalParams`, which is only called for sigs and defs. I suspect that if you change the signature of this function from taking a `Spec` to taking a `List[FormalParam]`, you should be able to generate the appropriate fparam error.

For the `ShadowedVar` error, you will need to explicitly check for shadowing on each formal parameter with the `shadowing` function. Look at `Expression.Let` for how this is done.

This requires some extra thought since this is the first time we have formal parameters in an environment that already contains other symbols.

Reply:
See new version just pushed. The tests pass, which is a good start!